uat/testing-features/src/main/java/com/aws/greengrass/steps/CloudWatchSteps.java

```diff
@@ -80,34 +100,107 @@ public void deleteLogGroup(String componentType, String componentName) {
      * created within the group.
      *
-     * @param componentType           The type of the component {GreengrassSystemComponent, UserComponent}
-     * @param componentName           The name of your component e.g. ComponentA, aws.greengrass.LogManager
-     * @param timeout                 Number of seconds to wait before timing out the operation
-     * @throws InterruptedException   {@link InterruptedException}
+     * @param componentType The type of the component {GreengrassSystemComponent, UserComponent}
+     * @param componentName The name of your component e.g. ComponentA, aws.greengrass.LogManager
+     * @param timeout       Number of seconds to wait before timing out the operation
+     * @throws InterruptedException {@link InterruptedException}
      */
 
-    @Then("I verify that it created a log group for component type {word} for component {word}, with streams within "
+    @Then("I verify that it created a log group of type {word} for component {word}, with streams within "
             + "{int} seconds in CloudWatch")
     public void verifyCloudWatchGroupWithStreams(String componentType, String componentName, int timeout) throws
-            InterruptedException {
+            Exception {
         SimpleDateFormat formatter = new SimpleDateFormat("yyyy/MM/dd", Locale.ENGLISH);
         formatter.setTimeZone(TimeZone.getTimeZone("UTC")); // All dates are UTC, not local time
 
-        String thingName = testContext.coreThingName();
-        String region = resourceContext.region().toString();
-
-        String logGroupName = String.format("/aws/greengrass/%s/%s/%s", componentType, region, componentName);
-        String logStreamNamePattern = String.format("/%s/thing/%s", formatter.format(new Date()), thingName);
+        String logGroupName = getLogGroupName(componentType, componentName);
+        String logStreamName = getLogStreamName();
 
         resources.create(CloudWatchLogStreamSpec
                 .builder()
                 .logGroupName(logGroupName)
-                .logStreamName(logStreamNamePattern)
+                .logStreamName(logStreamName)
                 .build());
 
-        LOGGER.info("Verifying log group {} with stream {} was created", logGroupName, logStreamNamePattern);
-        waitSteps.untilTrue(() -> doesStreamExistInGroup(logGroupName, logStreamNamePattern), timeout,
+        LOGGER.info("Verifying log group {} with stream {} was created", logGroupName, logStreamName);
+        boolean result = waitSteps.untilTrue(() -> doesStreamExistInGroup(logGroupName, logStreamName), timeout,
                 TimeUnit.SECONDS);
+
+        if (!result) {
+            throw new Exception(String.format("Failed to find log stream %s within group %s",
+                    logStreamName, logGroupName));
+        }
+    }
+
+
+    @Then("I verify {int} logs for {word} of type {word} have been uploaded to Cloudwatch within {int} seconds")
```

Comment:
I thought about doing that but then it would add a bunch of complexity given that at the time we check the files might be rotated or still being written to by the log generator + the added logic to read each and every file that matches the pattern etc etc.

This way was simpler given I could just ask it for what I expect given on how the Log Generator writes logs

Reply:
Agreed. We can also make use of scenario context variable in the steps like `<NoOfLogLines>` to make sure that we're checking for the same number of logs lines..